client/packages/common/src/authentication/api/hooks/useLogin.ts

```diff
@@ -70,11 +70,15 @@ export const getStore = async (
     item => item.username.toLowerCase() === userDetails?.username?.toLowerCase()
   );
 
-  if (mru?.store && stores?.some(store => store.id === mru?.store?.id)) {
+  if (
```

Comment:
I've had to change - this is possibly and edge case, but to be safe..

My most recently used store was disabled, then I logged in and was logged into it:
This is the `Opua` store after disabling, syncing and then logging out and in again:

<img width="411" alt="Screenshot 2024-06-24 at 11 36 37 AM" src="https://github.com/msupply-foundation/open-msupply/assets/9192912/ceda84d9-b113-40e2-8595-92321f12abf8">


I've filtered the store list, like so: 
```
export const getStore = async (
  userDetails?: Partial<UserNode>,
  mostRecentCredentials?: AuthenticationCredentials[]
) => {
  const defaultStore = userDetails?.defaultStore;
  const stores = userDetails?.stores?.nodes.filter(s => !s.isDisabled);
  const mru = mostRecentCredentials?.find(
    item => item.username.toLowerCase() === userDetails?.username?.toLowerCase()
  );

  if (
    mru?.store &&
    !mru.store.isDisabled &&
    stores?.some(store => store.id === mru?.store?.id && !store.isDisabled)
  ) {
    return stores.find(store => store.id === mru.store?.id) ?? mru.store;
  }

  if (!!defaultStore && !defaultStore.isDisabled) return defaultStore;

  return !!stores && stores?.length > 0 ? stores?.[0] : undefined;
};
```

in which case I think the `    !mru.store.isDisabled &&` line is no longer required. The check should be on the list of stores returned by the server, not the MRU which could be stored before the store was disabled

Reply:
[aeb8239](https://github.com/msupply-foundation/open-msupply/pull/4255/commits/aeb82392984fa7543f2c875dc6352e78d9046ea4)